lib/matplotlib/cbook/__init__.py

```diff
@@ -2075,3 +2081,14 @@ def _check_and_log_subprocess(command, logger, **kwargs):
             .format(command, exc.output.decode('utf-8')))
     logger.debug(report)
     return report
+
+
+def _check_enum(values, **kwargs):
+    """
+    For each *key, value* pair in *kwargs*, check that *value* is in *values*;
+    if not, raise an appropriate ValueError.
+    """
+    for k, v in kwargs.items():
+        if v not in values:
+            raise ValueError("{} must be one of {}, not {}"
```

Comment:
It's a good idea to state the offending value. However, it's hard to spot when trailing a long list:

`origin must be one of None, 'lower', 'upper', 'image', not 'org'`

Can we bring it close to the variable name, e.g. one of:

`Invalid value origin='org'. Supported values: None, 'lower', 'upper', 'image'`

`Invalid value 'org' for origin. Supported values: None, 'lower', 'upper', 'image'`

`'org' is not a valid value for origin. Supported values: None, 'lower', 'upper', 'image'`

Reply:
done